### PR TITLE
Do not morph links with non-self targets

### PR DESCRIPTION
**Pull request by alshakero**
Fixes: https://github.com/Palindrom/Palindrom/issues/73

**Please expand the bottom file's diff, GH decided to collapse it while it's important**.

### CODE REVIEW

File: src/palindrom-dom.js [hunk 1]
```diff
@@ -121,18 +123,20 @@ var PalindromDOM = (function() {
...
-    //while target.getAttribute("href") returns desired href (as string)
-    var href = target.href || target.getAttribute('href');
-
-    if (href && PalindromDOM.isApplicationLink(href)) {
-      event.preventDefault();
-      event.stopPropagation();
-      this.morphUrl(href);
-    } else if (target.type === 'submit') {
-      event.preventDefault();
+    var anchorTarget = target.target || target.getAttribute('target');
```
Comment (warpech) [inline src/palindrom-dom.js]: In what cases `var anchorTarget = target.target;` is not enough?

File: test/specs/dom/index.js [hunk 2]
```diff
@@ -102,127 +114,130 @@ if (typeof window !== "undefined") {
...
-        describe("should not intercept external links", function() {
-          it("full URL in the same host, different port", function() {
-            const port = window.location.port === "80" ||
-              window.location.port === ""
-              ? "8080"
-              : "80";
-            const href = window.location.protocol +
-              "//" +
+            expect(historySpy.callCount).to.equal(1);
+          });
```
Comment (warpech) [inline test/specs/dom/index.js]: Please add a test for empty `target` and no `target`.

File: src/palindrom-dom.js (hunk 1 above)
Comment (alshakero) [inline src/palindrom-dom.js]: I don't know. I thought `href` (two lines below) is written like this after a long debugging pain. So I did the same.

File: test/specs/dom/index.js (hunk 2 above)
Comment (alshakero) [inline test/specs/dom/index.js]: Sure

**Comment by alshakero** [comment]
> Do we have a convention to use single instead of double quotes?

I use [prettier](https://www.npmjs.com/package/prettier). Awesome experience tbh. And it got so popular that getting used to its style is a good asset. I'll note that in guidelines.

**Comment by tomalec** [comment]
> Do we have a convention to use single instead of double quotes?

We do follow Polymer convention, to use `'` in JS to ba able to use `"` for attributes in stringified HTML easier.

File: test/Sauce/Runner.js [hunk 3]
```diff
@@ -14,16 +14,16 @@ if (!username) {
     {
       browserName: "chrome",
       platform: "Windows 10",
-      version: "57.0",
+      version: "59.0",
```
Comment (tomalec) [inline test/Sauce/Runner.js]: Can't we use something like `"latest"`?

Comment (alshakero) [inline test/Sauce/Runner.js]: I guess not (https://wiki.saucelabs.com/display/DOCS/Platform+Configurator).

Comment (tomalec) [inline test/Sauce/Runner.js]: WCT set's it by not selecting the version at all https://github.com/Starcounter/starcounter-include/blob/master/wct.conf.json#L12 I would suspect it just propagates this config to sauce

https://github.com/saucelabs/ci-sauce/issues/3
> Though, we realized that current version is used when we use an empty string for "version" in Selenium's DesiredCapabilities.

https://wiki.saucelabs.com/display/DOCS/Test+Configuration+Options#TestConfigurationOptions-BrowserVersion
> If you want to use the latest stable version of Google Chrome or Firefox that Sauce supports, you can use "version": "latest". You can also use "version": "latest-1" or "version": "latest-2", etc. to request the next most recent versions of a browser. For example, if the latest stable version of Chrome is 48, you can request "latest-2" to use Chrome 46.

Comment (alshakero) [inline test/Sauce/Runner.js]: Then their own configurator is stupid -_-. Thanks for digging deeper.

**Comment by alshakero** [comment]
Is this good to go?

**Comment by alshakero** [comment]
@warpech thank you so much for those pre-weekend reviews. I really appreciate it.  I will enter the week excited. 

I'll test with KS before merging.

**Comment by warpech** [comment]
Ping
